File: implementations/python/lsf/parser/parser_options.py
"""Parser options for LSF."""
from dataclasses import dataclass


@dataclass
class ParserOptions:
    """Options for configuring LSF parser behavior."""
    
    max_input_size: int = 10 * 1024 * 1024  # 10MB default
    """Maximum allowed input size in bytes."""
    
    validate_type_hints: bool = True
    """Whether to validate type hints during parsing."""
    
    @classmethod
    def default(cls) -> 'ParserOptions':
        """Get default parser options."""
        return cls()

File: implementations/javascript/src/parser/parser-options.ts
/**
 * Options for configuring LSF parser behavior
 */
export interface ParserOptions {
  /**
   * Maximum allowed input size in bytes
   * @default 10485760 (10MB)
   */
  maxInputSize?: number;
  
  /**
   * Whether to validate type hints during parsing
   * @default true
   */
  validateTypeHints?: boolean;
}

/**
 * Default parser options
 */
export const defaultParserOptions: Required<ParserOptions> = {
  maxInputSize: 10 * 1024 * 1024, // 10MB
  validateTypeHints: true
};

/**
 * Merge user options with defaults
 */
export function mergeOptions(options?: ParserOptions): Required<ParserOptions> {
  return {
    ...defaultParserOptions,
    ...options
  };
}